pkg/sql/sessioninit/cache.go

```diff
@@ -318,16 +347,36 @@ func (a *Cache) GetDefaultSettings(
 				return nil
 			}
 
-			// Lookup the data outside the lock.
-			settingsEntries, err = readFromSystemTables(
-				ctx,
-				txn,
-				ie,
-				username,
-				databaseID,
+			// Lookup the data outside the lock, with at most one request in-flight
+			// for each user+database. The db_role_settings table version is also part
+			// of the request key so that we don't read data from an old version
+			// of the table.
+			ch, _ := a.populateCacheGroup.DoChan(
+				fmt.Sprintf("%s-%d-%d", username.Normalized(), databaseID, dbRoleSettingsTableVersion),
+				func() (interface{}, error) {
```

Comment:
the two usages of the cache look the same, including the select loop below - should we make this DoChan a method on `*Cache`?

Reply:
yeah, sounds good!